inference-engine/src/mkldnn_plugin/nodes/mkldnn_matrix_nms_node.cpp

```diff
@@ -352,9 +361,19 @@ void MKLDNNMatrixNmsNode::execute(mkldnn::stream strm) {
         }
     }
 
-    float* selectedOutputs = reinterpret_cast<float*>(getChildEdgesAtPort(NMS_SELECTED_OUTPUTS)[0]->getMemoryPtr()->GetPtr());
-    int* selectedIndices = reinterpret_cast<int*>(getChildEdgesAtPort(NMS_SELECTED_INDICES)[0]->getMemoryPtr()->GetPtr());
-    int* validOutputs = reinterpret_cast<int*>(getChildEdgesAtPort(NMS_VALID_OUTPUTS)[0]->getMemoryPtr()->GetPtr());
+    auto selectedOutputsMemPtr = getChildEdgesAtPort(NMS_SELECTED_OUTPUTS)[0]->getMemoryPtr();
+    auto selectedIndicesMemPtr = getChildEdgesAtPort(NMS_SELECTED_INDICES)[0]->getMemoryPtr();
+    auto validOutputsMemPtr = getChildEdgesAtPort(NMS_VALID_OUTPUTS)[0]->getMemoryPtr();
+
+    if (!m_outStaticShape) {
```

Comment:
add TODO
please see:

Reply:
TODO added.
If use 'isDynamicNode()', model has Nms->Convert->Multiply->Result and the inputs are dynamic which will result the 'isDynamicNode' is true:
   we should have static output shape because the [transformation](https://github.com/openvinotoolkit/openvino/blob/2b9fedfc772101f3fb5289670d3c393a730c547f/inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp#L321), but we will [call 'redefineDesc'](https://github.com/openvinotoolkit/openvino/blob/beaec484bbbf7ce1062f387449aed70bc31d961c/inference-engine/src/mkldnn_plugin/nodes/mkldnn_non_max_suppression_node.cpp#L230) try to change the output shape and not fill -1 because they will excute when ['isDynamicNode'==false](https://github.com/openvinotoolkit/openvino/blob/beaec484bbbf7ce1062f387449aed70bc31d961c/inference-engine/src/mkldnn_plugin/nodes/mkldnn_non_max_suppression_node.cpp#L255)

I think the problem is the output shape dynamic/static state is independent with the input shape state. If we will not support output static shape the logic can be simplified.